<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1053</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="DequeView" name="view">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>40</y>
      <width>471</width>
      <height>481</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="backBox">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>60</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="frontBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>60</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushBackButton">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>110</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Добавить в конец</string>
    </property>
   </widget>
   <widget class="QPushButton" name="popBackButton">
    <property name="geometry">
     <rect>
      <x>768</x>
      <y>170</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Удалить в конце</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushFrontButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>110</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Добавить в начало</string>
    </property>
   </widget>
   <widget class="QPushButton" name="popFrontButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>170</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Удалить в начале</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1053</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DequeView</class>
   <extends>QTextEdit</extends>
   <header>dequeview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
